docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,374 @@
+    other nodes in the cluster.
+  - Without a default route kube-apiserver does not start.
+- kubelet (part of kubelite)
+  - uses the default network interface to pick the node's InternalIP address.
+  - A default gateway greatly simplifies the process of setting up the
+    network feature.
+
+In case your air gap environment does not have a default gateway,
+you can add a dummy default route on interface eth0 using the following command:
+
+```bash
+ip route add default dev eth0
+```
+
+```{note} Confirm the name of your default network interface used for pod-to-pod
+    communication by running "ip a".
+```
+
+```{note} The dummy gateway will only be used by the Kubernetes services to 
+    know which interface to use,
+    actual connectivity to the internet is not required.
+    Ensure that the dummy gateway rule survives a node reboot.
+```
+
+#### (Optional) Network Requirement: Ensure proxy access
+
+Please skip this section, if you can't use an HTTP proxy (e.g. squid)
+with limited access to 
+image registries (e.g. docker.io, quay.io, rocks.canonical.com, etc).
+
+Ensure that all nodes can use the proxy to access the image registry.
+In this example we use squid as an http proxy.
+This set up uses http://squid.internal:3128 to access docker.io.
+
+Test the connectivity:
+
+```
+export https_proxy=http://squid.internal:3128
+curl -v https://registry-1.docker.io 
+```
+<!-- TODO: 404 on curl and https://registry-1.docker.io/v2 unauthorized -->
+
+Please refer to the next section `images` on how to use the HTTP proxy
+to allow limited access to image registries.
+
+## Prep 3: Images
+
+All workloads in a Kubernetes cluster are running as an OCI image.
+Kubernetes needs to be able to fetch these images and load them
+into the container runtime, in order to run workloads.
+For a Canonical Kubernetes deployment, you will need to fetch the images
+used by its features (network, dns, etc) as well as any images that are
+needed to run your workloads.
+
+The following options are presented in the order of
+increasing complexity of implementation.
+You may also find it helpful to combine these options for your scenario.
+
+If you already have a running cluster,
+you can list the images in use by running:
+  
+```bash
+sudo k8s kubectl get node -o template='{{ range .items }}{{ .metadata.name }}\
+{{":"}}{{ range .status.images }}{{ "\n- " }}{{ index .names 1 }}\
+{{ end }}{{"\n"}}{{ end }}'
+```
+
+### Images Option A: via an HTTP proxy
+
+In many cases, the nodes of the airgap deployment may not have direct access to
+upstream registries, but can reach them through the
+[use of an HTTP proxy][proxy].
+
+To configure the proxy, edit `/etc/squid/squid.conf` and set the appropriate
+allowed networks and domains. 
+
+Upon changes to the proxy configuration,
+restart both the squid proxy and the k8s snap with:
+
+```bash
+sudo systemctl restart squid
+```
+  
+```bash
+sudo snap restart k8s
+```
```

Comment:
there is more to configuring a squid proxy for dockerhub or a registry mirror, remove this as it is out of scope of this document